userdetails-cognito/src/main/groovy/au/org/ala/userdetails/CognitoUserService.groovy

```diff
@@ -504,42 +520,100 @@ class CognitoUserService implements IUserService {
 
     @Override
     void findScrollableUsersByUserName(String username, int maxResults, ResultStreamer resultStreamer) {
-        throw new NotImplementedException()
+        streamUserResults(resultStreamer, listUsers(username, null, maxResults))
     }
 
     @Override
     void findScrollableUsersByIdsAndRole(List<String> ids, String roleName, ResultStreamer resultStreamer) {
-        throw new NotImplementedException()
+
+        def things = ids.groupBy { it.isLong() }
+        def userIds = things[false]
+        def numberIds = things[true]
+
+        ListUsersInGroupRequest request = new ListUsersInGroupRequest().withUserPoolId(poolId)
+
+        request.groupName = roleName.split(jwtProperties.getRolePrefix())[1].toLowerCase()
+
+        def response = cognitoIdp.listUsersInGroup(request)
+        def users = response.users.findAll
+                {(!numberIds && !userIds) || numberIds?.contains(it.attributes.find{ attr -> attr.name == "name"}?.value) || userIds?.contains(it.username)}.stream()
+
+        def results = users.map { userType ->
+            cognitoUserTypeToUserRecord(userType, true)
+        }.toList()
+
+        streamUserResults(resultStreamer, results)
```

Comment:
Ditto above

Reply:
Shall I address this is in the clean up branch since there are few new changes here in clean up branch?

Reply:
Sounds good